src/renderer/components/ComposeDeployment.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * ************************************
+ *
+ * @module  ComposeDeployment.tsx
+ * @author Yevgeniy Skroznikov
+ * @date 3/11/20
+ * @description container for the title, the service info and the file open
+ *
+ * ************************************
+ */
+
+import React, { useState, useEffect } from 'react';
+import { FaUpload, FaDownload } from 'react-icons/fa';
+import { remote } from 'electron';
+import { 
+  runDockerComposeDeployment,
+  runDockerComposeKill,
+  runDockerComposeListContainer
+} from '../../common/runShellTasks';
+
+import {
+  FileOpen
+} from '../App.d';
+
+enum DeploymentStatus {
+    OpeningFile = 0,
+    NoFile,
+    Dead,
+    DeadError,
+    Checking,
+    Deploying,
+    Undeploying,
+    Warning,
+    Running,
+};
+
+type Props = {
+    currentFilePath: string;
+    fileOpen: FileOpen
+};
+
+const Deployment: React.FC<Props> = ({ currentFilePath, fileOpen }) => {
+  const [ deployState, setDeployState ] = useState(DeploymentStatus.NoFile);
+  const [ errorMessage, setErrorMessage ] = useState('');
+
+  useEffect(() => {
+    if(currentFilePath !== '') deployCheck();
+    else if(deployState !== DeploymentStatus.NoFile) setDeployState(DeploymentStatus.NoFile);
+  }, [currentFilePath]);
+
+  const deployCheck = () => {
+    if(deployState === DeploymentStatus.OpeningFile && currentFilePath !== ''){
+      deployCompose();
+    }
+    else {
+      setDeployState(DeploymentStatus.Checking)
+      runDockerComposeListContainer(currentFilePath)
+      .then((results: any) => {
+        if(results.error) {
+          setErrorMessage(results.error.message);
+          setDeployState(DeploymentStatus.DeadError);
+        } 
+        else if(results.out.split('\n').length > 3){
+          if(results.out.includes('Exit')) setDeployState(DeploymentStatus.Dead);
+          else setDeployState(DeploymentStatus.Running);
+        }
+        else setDeployState(DeploymentStatus.Dead);
+      });
+    }
+  };
+
+  const deployCompose = () => {
+    setDeployState(DeploymentStatus.Deploying)
+    runDockerComposeDeployment(currentFilePath)
+      .then((results: any) => { 
+        if(results.error) {;
+          setErrorMessage(results.error.message);
+          setDeployState(DeploymentStatus.DeadError);
+        }
+      else setDeployState(DeploymentStatus.Running);
+    })
+    .catch(err => console.log('err', err));
+  }
+
+  const deployKill = () => {
+    runDockerComposeKill(currentFilePath).then(() => setDeployState(DeploymentStatus.Dead));
+    setDeployState(DeploymentStatus.Undeploying);
+  }
+
+  const onErrorClick = () => {
+    const dialog = remote.dialog;
+    dialog.showErrorBox('Error Message:', errorMessage);
+  }
+
+  let title, onClick, icon = <FaUpload className="deployment-button" size={24} />;
```

Comment:
weird handling of all of this conditionals. all these ifstatements seem like you could use a switch or a map object...

Reply:
How would a switch statement handle OR conditionals?